### PR TITLE
Unique ops

### PR DESCRIPTION
**Pull request by gaga1313**
## Description
feat: Add `return_index` support to the `unique` operation

This PR introduces support for the `return_index` parameter within the `unique` operation across all supported backends.

Key Changes:
*   **`return_index` Functionality:**
    *   Implemented support for `return_index` in the following backends: `numpy`, `jax`, `torch`, `tensorflow`, and `openvino`.
    *   Indices returned by `return_index` are now padded with a value of `1` when the number of unique elements is less than the pre-defined size parameter.
    *   Added comprehensive test cases to validate the new `return_index` feature.

*   **Numpy Backend Bug Fix:**
    *   Fixed a bug in the `numpy` backend's `return_counts` index logic. The existing code incorrectly assumed `return_counts` would have the same shape as the input tensor, whereas it is always a 1-D array. This has been corrected.

## Contributor Agreement
**Please review our
[AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy)
and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/backend/torch/numpy.py [hunk 1]
```diff
@@ -2374,18 +2375,29 @@ def unique(
...
+    if return_index:
+        inverse = output[1]
+        perm = torch.arange(
+            inverse.size(0), dtype=inverse.dtype, device=inverse.device
+        )
+        inverse, perm = inverse.flip([0]), perm.flip([0])
+        dim = 0 if axis is None else axis
+        unique_indices = inverse.new_empty(values.size(dim)).scatter_(
+            dim, inverse, perm
+        )
```
[the remark this answers is not in the document]
Comment (gaga1313) [reply keras/src/backend/torch/numpy.py]: resolved!

File: keras/src/ops/numpy.py [hunk 2]
```diff
@@ -9725,6 +9736,8 @@ def unique(
         A tensor or a tuple of tensors.
         - `unique_values`: The unique values. Shape is `(size,)` if `size` is
           provided and `axis=None`.
+        - `unique_index` (optional): The indices of the first occurrence of each
+          unique element in `x`. Only provided if `return_index` is `True`.
```
Comment (hertschuh) [inline keras/src/ops/numpy.py:9740]: Remove, this feature did not get added.

And also remove `unique_values` above, this shouldn't have been there.

File: keras/src/ops/numpy_test.py [hunk 3]
```diff
@@ -7350,6 +7359,66 @@ def test_unique(self):
...
+        x = np.array([[1, 0], [0, 1], [1, 0], [2, 2], [0, 1]])
+        v, idx, inv, counts = knp.unique(
+            x,
+            axis=0,
+            return_index=True,
+            return_inverse=True,
+            return_counts=True,
+            size=5,
+            fill_value=-1,
+        )
```
Comment (hertschuh) [inline keras/src/ops/numpy_test.py:7406]: Can you also add a 2D test with `axis=-1`?

File: keras/src/ops/numpy_test.py [hunk 4]
```diff
@@ -3183,23 +3186,29 @@ def test_unique_symbolic_static(self):
...
 
-        v_axis0 = knp.unique(x, axis=0, size=3)
+        v_axis0, idx0 = knp.unique(x, return_index=True, axis=0, size=3)
         self.assertEqual(v_axis0.shape, (3, 4))
+        self.assertEqual(idx0.shape, (3,))
 
-        v_axis1 = knp.unique(x, axis=1, size=6)
+        v_axis1, idx1 = knp.unique(x, return_index=True, axis=1, size=6)
         self.assertEqual(v_axis1.shape, (2, 6))
+        self.assertEqual(idx1.shape, (6,))
```
Comment (hertschuh) [inline keras/src/ops/numpy_test.py:3211]: Can you also add a 2D case with `axis=-1`?

File: keras/src/backend/numpy/numpy.py
Comment (hertschuh) [inline keras/src/backend/numpy/numpy.py]: There should be a `canonicalize_axis` here to handle negative indices.

**Review by hertschuh — CHANGES_REQUESTED** [review]
Thanks!

A couple of comments:

File: keras/src/ops/numpy.py (hunk 2 above)
Comment (hertschuh) [reply keras/src/ops/numpy.py:9740]: Ignore this comment

Comment (gaga1313) [reply keras/src/ops/numpy.py:9740]: Resolved

File: keras/src/ops/numpy_test.py (hunk 4 above)
Comment (gaga1313) [reply keras/src/ops/numpy_test.py:3211]: resolved added new test

File: keras/src/ops/numpy_test.py (hunk 3 above)
Comment (gaga1313) [reply keras/src/ops/numpy_test.py:7406]: resolved: added new test

**Review by gaga1313 — COMMENTED** [review]
Resolved all comments

File: keras/src/backend/torch/numpy.py [hunk 5]
```diff
@@ -2374,20 +2375,34 @@ def unique(
...
         output = [output]
     else:
         output = list(output)
 
     values = output[0]
 
+    if axis is None:
+        dim = 0
+    else:
+        dim = axis + x.ndim if axis < 0 else axis
```
Comment (hertschuh) [inline keras/src/backend/torch/numpy.py]: Please use `canonicalize_axis(axis, x.ndim)` it's already imported in this file and it does validation.

File: keras/src/backend/numpy/numpy.py [hunk 6]
```diff
@@ -1726,21 +1727,25 @@ def unique(
...
         output = list(output)
 
     values = output[0]
 
     if size is not None:
-        dim = axis if axis is not None else 0
+        if axis is None:
+            dim = 0
+        else:
+            dim = axis + x.ndim if axis < 0 else axis
```
Comment (hertschuh) [inline keras/src/backend/numpy/numpy.py]: Please use `canonicalize_axis(axis, x.ndim)` it does some validation.

You'll need to add `from keras.src.backend.common.backend_utils import canonicalize_axis` at the top.

**Review by hertschuh — CHANGES_REQUESTED** [review]
Last nitpick:

**Review by hertschuh — APPROVED** [review]
Thanks!
